### PR TITLE
Product detail skeleton

### PR DESCRIPTION
**Pull request by seguyy**
Worked on Product and subcomponents; Style and Image etc. 
Implemented test suites for rendering. 
Improved Layout.
Added controllers, routers.

### CODE REVIEW

File: server/routes/products.js [hunk 1]
```diff
@@ -2,5 +2,7 @@ const router = require('express').Router();
 const controller = require('../controllers/products');
 
 router.get('/', controller.get);
```
Comment (beanjermin) [inline server/routes/products.js]: move '/' line to the bottom. I think David was saying that the more generic endpoint should be at the bottom and the more specific endpoints should be at the top

File: server/controllers/products.js [hunk 2]
```diff
@@ -11,7 +12,33 @@ module.exports = {
...
+      headers: {
+        authorization: API_TOKEN,
+      },
+    })
+      .then(({ data }) => res.json(data))
+      .catch((err) => {
+        console.log('There was a problem in the server retrieving product styles: ', err);
         res.sendStatus(404);
       });
   },
```
Comment (beanjermin) [inline server/controllers/products.js:44]: it looks like we both created a getStyles method. Looks almost exactly the same as mine so I guess it's up to you whether you want to keep this one or use mine

File: client/src/components/productDetails/Product.jsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        .catch((err) => {
+          console.log('There was a problem in the server retrieving specific product styles: ', err);
+        });
+    }
+  }
+
+  useEffect(() => {
+    getSpecificProduct();
+    getProductStyles();
+  }, [product]);
```
Comment (beanjermin) [inline client/src/components/productDetails/Product.jsx:43]: This looks good, but just a recommendation to look into axios.all and spread. It allows you to make multiple axios requests at once so you don't have to create two different functions like this!

**Review by beanjermin — CHANGES_REQUESTED** [review]

File: server/routes/products.js (hunk 1 above)
Comment (seguyy) [reply server/routes/products.js]: Changed it now

File: client/src/components/productDetails/Product.jsx (hunk 3 above)
Comment (seguyy) [reply client/src/components/productDetails/Product.jsx:43]: will look into all function!

**Comment by seguyy** [comment]
@beanjermin Hi Ben I changed the code accordingly, let me know how it looks!

**Review by beanjermin — DISMISSED** [review]
looks good!

**Review by davidguy3237 — APPROVED** [review]
I deleted the bundle.js files. Doesn't seem to be any noticeable conflicts otherwise. Should be good to go.

**Pull request merged**